src/unit-taxonomy-tags-drawer/UnitTaxonomyTagsDrawer.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo } from 'react';
+import {
+  Container,
+  CloseButton,
+} from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { useParams } from 'react-router-dom';
+import messages from './messages';
+import TaxonomyTagsCollapsible from './TaxonomyTagsCollapsible';
+import { extractOrgFromUnitId } from './utils';
+import {
+  useUnitTaxonomyTagsDataResponse,
+  useIsUnitTaxonomyTagsDataLoaded,
+  useUnitDataResponse,
+  useIsUnitDataLoaded,
+} from './api/hooks/selectors';
+import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded } from '../taxonomy/api/hooks/selectors';
+
+const UnitTaxonomyTagsDrawer = ({ intl }) => {
+  const { unitId } = useParams();
+
+  const org = extractOrgFromUnitId(unitId);
+
+  const useUnitData = () => {
+    const unitData = useUnitDataResponse(unitId);
+    const isUnitDataLoaded = useIsUnitDataLoaded(unitId);
+    return { unitData, isUnitDataLoaded };
+  };
+
+  const useUnitTaxonomyTagsData = () => {
+    const unitTaxonomyTagsData = useUnitTaxonomyTagsDataResponse(unitId);
+    const isUnitTaxonomyTagsLoaded = useIsUnitTaxonomyTagsDataLoaded(unitId);
+    return { unitTaxonomyTagsData, isUnitTaxonomyTagsLoaded };
+  };
+
+  const useTaxonomyListData = () => {
+    const taxonomyListData = useTaxonomyListDataResponse(org);
```

Comment:
I don't think the "taxonomy list" is actually necessary for the read-only view, though it's fine that you've implemented it because we'll need it in the editable version.

For the read-only version, the "get object tags" API returns the `name` (Taxonomy name) for each tag, which should be enough to display the view. (For the read-only version, we don't need to display taxonomies that don't yet have any tags.) And for displaying the ancestor tags, I'm working on an update to the API to make that possible, so it can be out of scope for now.

Reply:
In fact, I think we don't need to wait for the taxonomy list to load before displaying anything here. We could just call "get object tags" and immediately render whatever tags+taxonomies it returns, and at the same time (or later) we can call the "list taxonomies" and then for any taxonomies which don't yet have tags, we could append them to the list. No need to change this now though.

Reply:
Got it, thanks for the pointers, I'll keep this in mind when implementing the editing editable version.